services/provider/server/server.go

```diff
@@ -1229,13 +1229,45 @@ func (s *OCSProviderServer) getDesiredCephConnection(
 		return nil, fmt.Errorf("failed to extract monitor IPs from configmap %s: %v", monConfigMap, err)
 	}
 
+	var readAffinity *csiopv1a1.ReadAffinitySpec = nil
+	storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
+	if err != nil {
+		return nil, err
+	}
+	readAffinityOptions := util.GetReadAffinityOptions(storageCluster)
+
+	replica1Disabled := consumer.Annotations["ocs.openshift.io/disable-replica1"] == "true"
+
```

Comment:
Only this condition is needed `readAffinityOptions.Enabled && !replica1Disabled`
 and merge the labels always.

Reply:
@parth-gr done. Can you review?